### PR TITLE
PERF-3207 Create genny workload for running bulk non-targeted writes …

### PR DESCRIPTION
**Pull request by sphanse99**
…on an unsharded collection

Ticket: https://jira.mongodb.org/browse/PERF-3207

Tested locally. Evergreen patch successful: https://spruce.mongodb.com/version/633c9220e3c33104c2906b27/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: src/workloads/sharding/UpdateOneWithoutShardKey.yml
Comment (jz1242) [inline src/workloads/sharding/UpdateOneWithoutShardKey.yml:4]: I think this description should be of what the workload is doing, not really what the meta goal is. That we can put in the ticket or something.

Comment (jz1242) [reply src/workloads/sharding/UpdateOneWithoutShardKey.yml:4]: Also we can technically run multiple updateOnes in a single batch.

Comment (sphanse99) [reply src/workloads/sharding/UpdateOneWithoutShardKey.yml]: Done. I think it's better to run as single update commands to compare latency linearly, so keeping thread size as 1.

**Comment by sphanse99** [comment]
Workload runs successfully locally. In progress evg: https://spruce.mongodb.com/version/6349c2a93627e03ea92bfcd8/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml:1]: If you want to also include the sharded workload we might want to change the ticket description, and file a PERF ticket to enable the sharded workload later.

File: src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml:16]: I dont think this balancer wait is necessary since we dont have more than 1 shard now.

File: src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml:36]: same here we don't need to enable the balancer.

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml:13]: Same in this workload, the balancer isn't necessary and will needlessly slow down the workload.

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml:4]: [nit] findAndModifyOne -> findAndModify (it only does a single op)

File: src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml:5]: [nit] findAndModifyOne -> findAndModify (it only does a single op)

File: src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml:127]: Is this going to stay commented?

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml:111]: I'm not sure if these variants are all single shard variants. Can you check with server testing to make sure? I think also https://github.com/10gen/dsi/tree/master/configurations/mongodb_setup is where all the configurations are defined.

**Review by jz1242 — CHANGES_REQUESTED** [review]
Looks good! Couple of comments.

**Review by sphanse99 — COMMENTED** [review]
New Evg Patch: https://spruce.mongodb.com/version/634f15dd1e2d173bf13df272/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: src/workloads/sharding/UpdateOneWithoutShardKey.yml
Comment (jz1242) [reply src/workloads/sharding/UpdateOneWithoutShardKey.yml]: Oh I meant like multiple writes in a batch, not multiple threads. Here is some documentation on the CrudActor: https://github.com/mongodb/genny/blob/master/src/workloads/docs/CrudActor.yml

File: src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml
Comment (jz1242) [reply src/workloads/sharding/WriteOneWithoutShardKeyShardedCollection.yml]: I'm not 100 percent sure if it tests on non major releases. This feature won't be on 6.0 it's only available on master.

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml:56]: Could probably remove the UnshardedCollection suffix since it's already a part of the unsharded collection genny test and the name seems verbose same for the ones below.

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (jz1242) [inline src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml:10]: Since this is an unsharded collection test with only unsharded collections, we can probably remove the Unsharded prefix and just keep it as collection so as to not be so verbose. Same as above.

**Review by jz1242 — CHANGES_REQUESTED** [review]
Looks good last comments.

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (sphanse99) [reply src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml]: Good catch. I spoke to @aggrand and it seems `OutcomeCounts` is a method used internally by CrudActor tests but is not visible to other tests so this test isn't/can't run this check. They shared it is standard not to verify that documents were updated and seeing the performance statistics on t2 / evergreen can act as confirmation that the writes are completing. 

So, I am removing the `OutcomeCounts` step in the phase.

Comment (aggrand) [reply src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml]: That should play directly into `counters.n` documented here: https://github.com/10gen/performance-tooling-docs/blob/main/getting_started/intrarun_data_generation.md#client-side-intra-run-data

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (aggrand) [inline src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml:44]: If you did want more concurrent threads, you might be able to target separate docs that are filtered on ActorId or an expression thereof like [here](https://github.com/mongodb/genny/blob/58901acf3ed1a4dd41f55f41f1d46689ff804e51/src/workloads/docs/Generators.yml#L126), I'm pretty sure those are actor-thread unique. Up to you on that!

**Review by aggrand — COMMENTED** [review]
I can't comment much on the system under test or value of the workload, but regarding genny usage the syntax seems good! Feel free to hit me up with any specific questions I can add

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (sphanse99) [reply src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml]: Thanks for this context!

File: src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml
Comment (sphanse99) [reply src/workloads/sharding/WriteOneWithoutShardKeyUnshardedCollection.yml:44]: Thanks for the suggestion. Jason is OOO right now, so just waiting on his final approval but good to know that the genny syntax is looking good.

**Review by jz1242 — APPROVED** [review]
LGTM!

**Comment by sphanse99** [comment]
evergreen merge

**Pull request merged**
